auth/generic/provider.go

```diff
@@ -36,7 +36,10 @@ import (
 const ProviderName = "generic"
 
 // Provider implements the auth.Provider interface for generic authentication.
-type Provider struct{ Implementation }
+type Provider struct {
+	Client client.Client
+	Implementation
+}
 
 // GetName implements auth.RESTConfigProvider.
 func (p Provider) GetName() string {
```

Comment:
Please remove the `p` receiver where it is not needed :pray: 

```suggestion
func (Provider) GetName() string {
```